sway-core/src/semantic_analysis/ast_node/expression/intrinsic_function.rs

```diff
@@ -885,9 +885,10 @@ fn type_check_shift_binary_op(
         handler,
         ctx.by_ref()
             .with_help_text("Incorrect argument type")
-            .with_type_annotation(return_type),
+            .with_type_annotation(engines.te().insert(engines, TypeInfo::Numeric)),
```

Comment:
This doesn't look right, we should be type checking the intrinsics. With this code you could have `let _: u32 = 1u64 << 1u64;` which is incorrect.

What's the reason to relax it here?

Reply:
You are right, This is wrong. Fixed.